lib/charms/layer/apache_zeppelin.py

```diff
@@ -13,15 +24,37 @@ class Zeppelin(object):
 
     :param DistConfig dist_config: The configuration container object needed.
     """
+    @classmethod
+    def get(cls):
+        """
+        Dynamically instantiate this or a subclass, to allow other layers
+        to override portions of this implementation.
+        """
+        impl = layer.options('apache-zeppelin')['implementation_class']
+        module_name, cls_name = impl.rsplit('.', 1)
+        cls = getattr(import_module(module_name), cls_name)
+        return cls()
```

Comment:
Hmmm ... I think that I'd rather see one of the following:

1) Override `__new__` with pretty much exactly this logic. That would allow people to just call Zeppelin(), without also calling something that looks like a Java getter :-)  You then have to explain the magic in a doc, but you have to explain it, anyway.

2) Do the logic you do here, but in the import line in the reactive handlers. If the implementation_class config isn't set, you just import Zeppelin. Otherwise, you import_module the thing in the implementation_class config.

3) Make people just write a custom Zeppelin layer for a custom Zeppelin subclass. I like this one the most, but I get why you didn't want to do it :-)